src/SSHLibrary/library.py

```diff
@@ -963,12 +966,12 @@ def read_command_output(self, return_stdout=True, return_stderr=False,
                                            return_rc)
         try:
             stdout, stderr, rc = self.current.read_command_output()
-        except SSHClientException, msg:
+        except SSHClientException as msg:
             raise RuntimeError(msg)
         return self._return_command_output(stdout, stderr, rc, *opts)
 
     def _legacy_output_options(self, stdout, stderr, rc):
-        if not isinstance(stdout, basestring):
+        if (not is_string(stdout)) and (not is_bytes(stdout)):
             return stdout, stderr, rc
```

Comment:
The new expression looks pretty complicated and potentially buggy. Couldn't this just be `not (is_string(stdout) or is_bytes(stdout))`?

Also, the existing code looks strange. When would stdout not be Unicode or bytes??

Reply:
it is called by keyword "read_command_output"

```
def read_command_output(self, return_stdout=True, return_stderr=False,
                            return_rc=False):
```

it is bool be default.